streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamJoinWindow.java

```diff
@@ -28,13 +28,13 @@
     private final String windowName;
 
     /**
-     * @throws TopologyException if retention period of the join window is less than expected
+     * @throws StreamsException if retention period of the join window is less than expected
      */
     KStreamJoinWindow(String windowName, long windowSizeMs, long retentionPeriodMs) {
         this.windowName = windowName;
 
         if (windowSizeMs > retentionPeriodMs)
-            throw new TopologyException("The retention period of the join window "
+            throw new StreamsException("The retention period of the join window "
```

Comment:
I think, that a `TopologyException` is correct here. `KStreamJoinWindows` is called before `KafkaStreams#start()`. Or do I miss anything?

Reply:
+1

Reply:
+1